go/store/prolly/map.go

```diff
@@ -168,9 +168,54 @@ func (m Map) IterAll(ctx context.Context) (MapRangeIter, error) {
 	return m.IterRange(ctx, rng)
 }
 
+// IterOrdinalRange returns a MapRangeIter for the ordinal range beginning at |start| and ending before |stop|.
+func (m Map) IterOrdinalRange(ctx context.Context, start, stop uint64) (MapRangeIter, error) {
+	if stop == start {
+		return emptyIter{}, nil
+	} else if stop < start {
+		return nil, fmt.Errorf("invalid ordinal bounds (%d, %d)", start, stop)
+	}
+
+	lo, err := newCursorAtOrdinal(ctx, m.ns, m.root, start)
+	if err != nil {
+		return nil, err
+	}
+
+	hi, err := newCursorAtOrdinal(ctx, m.ns, m.root, stop)
+	if err != nil {
+		return nil, err
+	}
+
+	return &prollyRangeIter{
+		curr: lo,
+		stop: hi,
+	}, nil
+}
+
 // IterRange returns a MutableMapRangeIter that iterates over a Range.
 func (m Map) IterRange(ctx context.Context, rng Range) (MapRangeIter, error) {
-	return m.iterFromRange(ctx, rng)
+	if rng.isPointLookup(m.keyDesc) {
+		return m.pointLookupFromRange(ctx, rng)
+	} else {
+		return m.iterFromRange(ctx, rng)
+	}
+}
+
+func (m Map) pointLookupFromRange(ctx context.Context, rng Range) (*pointLookup, error) {
```

Comment:
Bothers me a little that we're doing this IO effectively at compile time rather than at runtime. This is a very fat constructor. It would be better to move the IO into the call to `iter.Next`

Reply:
That's a good shout, but I'm not sure it's true in practice. 
Essentially all table access is wrapped in a partition iterator which lazily constructs `RowIter`s for table scans, range queries, and joins. After the partitions row iter is requested, we typically perform IO eagerly.